src/common/transformations/src/transformations/op_conversions/einsum_decomposition.cpp

```diff
@@ -243,6 +350,99 @@ ov::Output<ov::Node> unsqueeze_input(const ov::Output<ov::Node>& input_node,
     return unsqueeze->output(0);
 }
 
+/// \brief Broadcasts and merges two shapes using specified broadcasting rules.
+///
+/// This function takes two  shapes (shapes_lhs and shapes_rhs) and attempts to broadcast
+/// and merge them into a single shape using NumPy and bidirectional broadcasting rules. The resulting
+/// broadcasted shape is returned as an OutputVector. If one of the input vectors is empty, the other
+/// vector is returned as is.
+///
+/// \param shapes_lhs A single element vector containing the left-hand side shape to be broadcasted or empty.
+/// \param shapes_rhs A single element vector containing the right-hand side shape to be broadcasted or empty.
+/// \param subgraph_nodes A vector to which the nodes created during the broadcasting process are added.
+/// \return An OutputVector containing the broadcasted and merged shape. If one of the input vectors is empty,
+///         the other vector is returned.
+///
+ov::OutputVector broadcast_merge_shapes(ov::OutputVector& shapes_lhs,
+                                        ov::OutputVector& shapes_rhs,
+                                        ov::NodeVector& subgraph_nodes) {
+    ov::OutputVector broadcasted_shape_nodes{};
+    // OutputVector is either empty or contains a single shape
+    if (shapes_lhs.size() == 1 && shapes_rhs.size() == 1) {
+        auto const_1 = ov::op::v0::Constant::create(ov::element::Type_t::i64, ov::Shape{1}, {1});
+        auto tmp_const_of_lhs_shp =
+            std::make_shared<ov::op::v3::Broadcast>(const_1, shapes_lhs[0], ov::op::BroadcastType::NUMPY);
+        auto tmp_const_of_broadcasted_shp =
+            std::make_shared<ov::op::v3::Broadcast>(tmp_const_of_lhs_shp,
+                                                    shapes_rhs[0],
+                                                    ov::op::BroadcastType::BIDIRECTIONAL);
```

Comment:
you create temporal tensor of the full size in order to compute the resulted shape. It is memory consuming used just for computing shape. We can operate only on shapes here.
We can use alignenment by rank:
1. compute maximum rank
2. append ones to both shapes to have this maximum rank
3. compute maximum from both aligned shapes

Reply:
Changed to just Maixmum op. Broadcasts were initial workarounds, however with addition of functions that unsqueeze ellipsis to required rank maximum should be sufficient